src/PriceOracle.sol

```diff
@@ -117,29 +120,35 @@ contract PriceOracle is AccessControl {
             uint32 _currentPrice = currentPrice;
             uint32 _minimumPrice = minimumPrice;
             uint32 _priceBase = priceBase;
+            uint64 _currentPriceUpscaled = currentPriceUpscaled;
 
             // Set the number of rounds that were skipped
             uint64 skippedRounds = currentRoundNumber - lastAdjustedRound - 1;
 
             // We first apply the increase/decrease rate for the current round
-            uint32 ir = increaseRate[usedRedundancy];
-            _currentPrice = (ir * _currentPrice) / _priceBase;
+            uint32 _changeRate = changeRate[usedRedundancy];
+            _currentPriceUpscaled = (_changeRate * _currentPriceUpscaled) / _priceBase;
 
             // If previous rounds were skipped, use MAX price increase for the previous rounds
             if (skippedRounds > 0) {
-                ir = increaseRate[0];
+                _changeRate = changeRate[0];
                 for (uint64 i = 0; i < skippedRounds; i++) {
-                    _currentPrice = (ir * _currentPrice) / _priceBase;
+                    _currentPriceUpscaled = (_changeRate * _currentPriceUpscaled) / _priceBase;
                 }
             }
 
+            _currentPrice = uint32(_currentPriceUpscaled) >> 10;
```

Comment:
you cast it to `uint32` then shift it? it should be the other way around

Reply:
As in the comment above, _currentPriceUpscaled is max 8 digits so when price raises 100x then we could have a problem. So it doesnt metter in this case, nothing would be lost, but I do agree its kind of better to have first bitshift then downcast.

If you wonder why did I use uint64 on other place it is the multiplication that has this numbers actually
 uint64 res1 = (524297 * 24576000) / 524288;
 
and in this case at one point it goes over uint32, so not to get problems I used uint64.